source/Engine.cpp

```diff
@@ -1633,6 +1639,36 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 
 
 
+// Changing the players flagship during flight requires an update of all active
+// data that references the players flagship.
+void Engine::ChangePlayerFlagship(const shared_ptr<Ship> oldFlagship, shared_ptr<Ship> &newFlagship)
```

Comment:
To my understanding, ship instances are all shared pointers. You only see const references when dealing with uninstantiated ships (e.g. ship definitions in shipyards).

Reply:
Passing `oldFlagship` as a reference to a shared pointer will result in having a reference to the flagshipPtr in PlayerInfo. If the flagshipPtr in PlayerInfo then changes by player.setFlagship(...) on line 1657, then the comparison on line 1633 would refer to the new flagship instead of the old flaghip (since it would be a reference to that pointer, not a pointer on its own).
So no, the oldFlagship really needs to be a pointer and not a reference to a pointer. For the newFlagship I also feel that I better change the code to use the shared_ptr as pointer not as a reference.

Reply:
oh oops sorry.

Reply:
I also used a reference here originally and spent quite a bit of time debugging on why the pointer was not correct. :-)